src/provider/HOCs/withVariableContext.js

```diff
@@ -53,4 +43,20 @@ export function withVariableContext(Component) {
 
     return <Component { ...props } variables={ variables }></Component>;
   };
-}
\ No newline at end of file
+}
+
+// helpers //////////
+
+function useServiceIfAvailable(service, fallback) {
+  const resolved = useService(service, false);
+
+  if (!resolved) {
+    return fallback;
+  }
+
+  return resolved;
+}
+
+const fallbackResolver = {
+  getVariablesForElement: bo => getVariablesForElement(bo)
+};
```

Comment:
This made me wonder if it's not a bug. `const` and `let` variables are not hoisted, so theoretically if one calls `withVariableContext` immediately after import, the variable may throw error of access before initialisation. I guess it's not a problem since the CI is passing, but I'd rather expect such defaults to be declared on top of the file.

Reply:
I will do it.